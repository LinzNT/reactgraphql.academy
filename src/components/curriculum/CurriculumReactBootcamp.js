import React from 'react'
import Link from '../navigation/Link'
import { H2Ref, H4 } from '../text'
import Section, { curriedToggleNavigateTo } from './CurriculumSection'
import { Col, Row } from '../layout/Grid'
import ES6Session from './sessions/ES6Session'
import ReactJS101Session from './sessions/ReactJS101Session'
import ThinkingInReactSession from './sessions/ThinkingInReactSession'
import RoutingAndDataFetchingSession from './sessions/RoutingAndDataFetchingSession'
import ReactFundamentalsRecapSession from './sessions/ReactFundamentalsRecapSession'
import FormsAndAuthSession from './sessions/FormsAndAuthSession'
import StylingInReactSession from './sessions/design/StylingInReactSession'
import IntroReduxSession from './sessions/IntroReduxSession'
import TestingIntroSession from './sessions/TestingIntroSession'
import TestingInReactSession from './sessions/TestingInReactSession'
import E2ESession from './sessions/E2ESession'
import HoCsRenderPropsStateReducerSession from './sessions/HoCsRenderPropsStateReducerSession'
import ReactPerformanceSession from './sessions/ReactPerformanceSession'
import Hackathon from './sessions/Hackathon'
import AdvancedReduxSession from './sessions/AdvancedReduxSession'
import { Li } from '../layout/Ul'

import { LinkButton } from '../buttons'
import SectionCTA from './SectionCTA'
import { REACT_BOOTCAMP } from '../../config/data'
import selectCurriculumLayout, { LIST_TWO_COL } from './selectCurriculumLayout'

const CurriculumReactBootcamp = ({
  showTitle = true,
  layout,
  enableToggle,
  isOpen,
  toggleNavigateTo = `/react/curriculum?tab=${REACT_BOOTCAMP}`,
  marketingCard = null,
  showLinkToCurriculum = true,
  trainings,
  showHackathon = false,
}) => {
  const toggleNavigateToSection = curriedToggleNavigateTo(toggleNavigateTo)
  const type = REACT_BOOTCAMP
  const commonProps = {
    enableToggle,
    toggleNavigateTo: toggleNavigateToSection,
    type,
    isOpen,
  }
  const firstHalf = (
    <React.Fragment>
      <H4>Course outline:</H4>
      <Section
        {...commonProps}
        title="Evening pre-bootcamp (optional)"
        name="day0"
        subTitle="React 101 and JS fundamentals"
      >
        <ReactJS101Session />
      </Section>
      <Section
        {...commonProps}
        title="React Bootcamp Day 1"
        name="day1"
        subTitle="Modern JavaScript, Thinking in React, Routing & Data Fetching"
      >
        <ES6Session title="Modern JavaScript" />
        <ThinkingInReactSession title="Thinking in React" />
        <RoutingAndDataFetchingSession title="Routing and Data Fetching" />
      </Section>
      <Section
        {...commonProps}
        title="React Bootcamp Day 2"
        name="day2"
        subTitle="Forms, Authentication, Styling in React"
      >
        <FormsAndAuthSession title="Forms and Authentication" />
        <ReactFundamentalsRecapSession
          title="React Fundamentals recap, build a React app from scratch on your own to
          consolidate:"
        />
        <StylingInReactSession title="Styling in React" />
      </Section>
      {marketingCard}
      {showLinkToCurriculum && showHackathon && (
        <SectionCTA>
          <LinkButton to={`/react/curriculum?tab=${REACT_BOOTCAMP}`}>
            Full curriculum
          </LinkButton>
        </SectionCTA>
      )}
    </React.Fragment>
  )
  const secondHalf = (
    <React.Fragment>
      <Section
        {...commonProps}
        title="React Bootcamp Day 3"
        name="day3"
        subTitle="Redux Fundamentals, Advanced Redux, and FP"
      >
        <IntroReduxSession title="Redux Fundamentals" />
        <AdvancedReduxSession title="Advanced Redux" />
      </Section>
      <Section
        {...commonProps}
        title="React Bootcamp Day 4"
        name="day4"
        subTitle="Advanced React patterns, Hooks, and performance"
      >
        <HoCsRenderPropsStateReducerSession title="Functional Programming & Advanced React patterns" />
        <ReactPerformanceSession title="Performance" />
      </Section>
      <Section
        {...commonProps}
        title="React Bootcamp Day 5"
        name="day5"
        subTitle="Real-world Testing in React"
      >
        <TestingIntroSession title="Testing Foundation in JS" />
        <TestingInReactSession title="Testing in React" />
        <E2ESession title="End-to-End Testing" />
      </Section>
      {showHackathon ? (
        <Section
          type={type}
          enableToggle={true}
          title="React Bootcamp Day 6 (optional)"
          name="day6"
          subTitle="GraphQL and final project"
        >
          <Hackathon />
        </Section>
      ) : (
        <SectionCTA>
          <LinkButton to={`/react/curriculum?tab=${REACT_BOOTCAMP}`}>
            Full curriculum
          </LinkButton>
        </SectionCTA>
      )}
    </React.Fragment>
  )

  return (
    <React.Fragment>
      {showTitle ? (
        <Row>
          <Col lg={10} lgOffset={layout !== LIST_TWO_COL ? 1 : 0}>
            <H2Ref>
              React Bootcamp Curriculum{' '}
              <Link to="#curriculum" name="curriculum">
                #
              </Link>
            </H2Ref>
          </Col>
        </Row>
      ) : (
        ''
      )}
      {selectCurriculumLayout({
        firstHalf,
        secondHalf,
        layout,
        type,
        trainings,
      })}
    </React.Fragment>
  )
}

export const TargetAudienceList = () => (
  <React.Fragment>
    <Li></Li>
  </React.Fragment>
)

export const LearningObjectives = () => (
  <React.Fragment>
    <Li>
      Understand the core principles and libraries to build production-ready
      React applications using:{' '}
      <code>
        react, react-router, styled-components, styled-system, storybook, redux,
        react-redux
      </code>
    </Li>
    <Li>
      Be able to develop and test complex and reliable React applications:{' '}
      <code>enzyme, jest</code>
    </Li>
    <Li>
      Comprehend modern front-end JavaScript:{' '}
      <code>Functional Programming, Webpack</code>
    </Li>
    <Li>
      Understand the best practices and patterns for building real-world
      performant React applications
    </Li>
  </React.Fragment>
)

CurriculumReactBootcamp.LearningObjectives = LearningObjectives
CurriculumReactBootcamp.TargetAudienceList = TargetAudienceList

export default CurriculumReactBootcamp
