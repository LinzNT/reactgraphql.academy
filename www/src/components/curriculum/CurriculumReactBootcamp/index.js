import React from 'react'
import ES6Session from '../sessions/ES6Session'
import ThinkingInReactSession from '../sessions/ThinkingInReactSession'
import RoutingAndDataFetchingSession from '../sessions/RoutingAndDataFetchingSession'
import ReactFundamentalsRecapSession from '../sessions/ReactFundamentalsRecapSession'
import FormsAndAuthSession from '../sessions/FormsAndAuthSession'
import StylingInReactSession from '../sessions/StylingInReactSession'
import HooksSession from '../sessions/HooksSession'
import IntroReduxSession from '../sessions/IntroReduxSession'
import ReactGraphQLIntroSession from '../sessions/ReactGraphQLIntroSession'
import TestingIntroSession from '../sessions/TestingIntroSession'
import TestingInReactSession2 from '../sessions/TestingInReactSession2'
import AdvancedReactPatternsSession from '../sessions/AdvancedReactPatternsSession'
import AdvancedHooksAndPerfSession from '../sessions/AdvancedHooksAndPerfSession'
import FundamentalsFinalProject from '../sessions/FundamentalsFinalProject'
import DesignSystemSession from '../sessions/DesignSystemSession'

import {
  FULL_TIME,
  TECH_REACT,
  TRAINING_TYPE_FULL_CURRICULUM,
  REACT_BOOTCAMP_ID,
} from '../../../config/data'
import Curriculum, { renderSection } from '../Curriculum'

export const sessionsFirstHalf = [
  {
    subTitle: 'Thinking in React, Modern JavaScript, Routing & Data Fetching',
    comps: [ThinkingInReactSession, ES6Session, RoutingAndDataFetchingSession],
  },
  {
    subTitle: 'Forms, Authentication, and Hooks',
    comps: [FormsAndAuthSession, HooksSession, ReactFundamentalsRecapSession],
  },
  {
    subTitle: 'Redux Fundamentals, deployment to production',
    comps: [IntroReduxSession, FundamentalsFinalProject],
  },
]
export const sessionsSecondHalf = [
  {
    subTitle: 'Advanced React patterns and performance',
    comps: [AdvancedReactPatternsSession, AdvancedHooksAndPerfSession],
  },
  {
    subTitle: 'GraphQL 101 & Real-World Testing in React',
    comps: [
      ReactGraphQLIntroSession,
      TestingIntroSession,
      TestingInReactSession2,
    ],
  },
  {
    subTitle: 'Building a UI component library',
    comps: [StylingInReactSession, DesignSystemSession],
  },
]

const trainingInstanceTypeName = FULL_TIME
const tech = TECH_REACT
const trainingType = TRAINING_TYPE_FULL_CURRICULUM
const trainingId = REACT_BOOTCAMP_ID

const CurriculumReactBootcamp = ({
  toggleNavigateTo = `/react/curriculum?tab=${trainingId}${trainingInstanceTypeName}`,
  training,
  section = {},
  ...rest
}) => {
  const sectionProps = {
    ...section,
    toggleNavigateTo,
    trainingInstanceTypeName,
    tech,
    trainingType,
    trainingId,
  }
  const renderSectionArgs = {
    training,
    sectionProps,
  }

  return (
    <Curriculum
      title="React Redux Bootcamp Outline"
      trainingId={trainingId}
      tech={tech}
      trainingType={trainingType}
      trainingInstanceTypeName={trainingInstanceTypeName}
      curriculumTo={toggleNavigateTo}
      {...rest}
      firstHalf={sessionsFirstHalf.map(renderSection(renderSectionArgs))}
      secondHalf={sessionsSecondHalf.map(
        renderSection({
          ...renderSectionArgs,
          initialDayOffset: sessionsFirstHalf.length,
        })
      )}
    />
  )
}

export default CurriculumReactBootcamp
