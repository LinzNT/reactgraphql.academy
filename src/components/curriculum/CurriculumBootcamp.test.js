import React from 'react'
import { shallow } from 'enzyme'
import { shallowToJson } from 'enzyme-to-json'

import CurriculumReactBootcamp from './CurriculumReactBootcamp'

describe('<CurriculumReactBootcamp />', () => {
  it('should render component', () => {
    const wrapper = shallow(<CurriculumReactBootcamp />)
    expect(shallowToJson(wrapper)).toMatchSnapshot()
  })
})
