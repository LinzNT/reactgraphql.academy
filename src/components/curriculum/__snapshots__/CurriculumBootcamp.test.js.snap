// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CurriculumBootcamp /> should render component 1`] = `
<Fragment>
  <Row>
    <Styled(Col)
      lg={10}
      lgOffset={1}
    >
      <Styled(styled.h2)>
        React Bootcamp Curriculum
         
        <Link
          name="curriculum"
          to="#curriculum"
        >
          #
        </Link>
      </Styled(styled.h2)>
    </Styled(Col)>
  </Row>
  <Row>
    <Styled(Col)
      lg={5}
      lgOffset={1}
      md={6}
    >
      <withRouter(CurriculumSection)
        name="day1"
        subTitle="React 101 and JS fundamentals"
        title="React Bootcamp Day 1 (half day)"
        toggleNavigateTo={[Function]}
        type="React bootcamp"
      >
        <ReactJS101Session />
      </withRouter(CurriculumSection)>
      <withRouter(CurriculumSection)
        name="day2"
        subTitle="Modern JavaScript, Thinking in React, Routing & Data Fetching"
        title="React Bootcamp Day 2"
        toggleNavigateTo={[Function]}
        type="React bootcamp"
      >
        <ES6Session
          title="Modern JavaScript"
        />
        <ThinkingInReactSession
          title="Thinking in React"
        />
        <RoutingAndDataFetchingSession
          title="Routing and Data Fetching"
        />
      </withRouter(CurriculumSection)>
      <withRouter(CurriculumSection)
        name="day3"
        subTitle="Forms, Authentication, Styling in React"
        title="React Bootcamp Day 3"
        toggleNavigateTo={[Function]}
        type="React bootcamp"
      >
        <FormsAndAuthSession
          title="Forms and Authentication"
        />
        <ReactFundamentalsRecapSession
          title="React Fundamentals recap, build a React app from scratch on your own to consolidate:"
        />
        <StylingInReactSession
          title="Styling in React"
        />
      </withRouter(CurriculumSection)>
      <withRouter(CurriculumSection)
        name="day4"
        subTitle="Redux, and Testing Principles"
        title="React Bootcamp Day 4"
        toggleNavigateTo={[Function]}
        type="React bootcamp"
      >
        <IntroReduxSession
          title="Introduction to Redux"
        />
        <TestingIntroSession
          title="Testing Principales"
        />
      </withRouter(CurriculumSection)>
      <withRouter(CurriculumSection)
        name="day5"
        subTitle="Functional Programming, Advanced React patterns I, and GraphQL"
        title="React Bootcamp Day 5"
        toggleNavigateTo={[Function]}
        type="React bootcamp"
      >
        <HoCsRenderPropsStateReducerSession
          title="Functional Programming & Advanced React patterns I"
        />
        <GraphQLSession
          title="GraphQL and Apollo client"
        />
      </withRouter(CurriculumSection)>
      <withRouter(CurriculumSection)
        name="day6"
        subTitle="Real-world Testing in React, Advanced React Patterns II, and SSR"
        title="React Bootcamp Day 6"
        toggleNavigateTo={[Function]}
        type="React bootcamp"
      >
        <TestingInReactSession
          title="Testing in React"
        />
        <CompoundCompAndContextSession
          title="Advanced React patterns to create even more reusable UIs"
        />
        <ServerSideRenderingSession
          title="Server Side Rendering (SSR)"
        />
      </withRouter(CurriculumSection)>
      <withRouter(CurriculumSection)
        name="day7"
        subTitle="Real-world Project"
        title="React Bootcamp Day 7"
        toggleNavigateTo={[Function]}
        type="React bootcamp"
      >
        <Hackathon
          title="Last day real-world React challenge. We'll implement an app in teams from scratch"
        />
      </withRouter(CurriculumSection)>
      <styled.div>
        <LinkButton
          secondary={true}
          to="/curriculum?tab=React bootcamp"
          trackUserBehaviour={[Function]}
        >
          Full curriculum &gt;&gt;
        </LinkButton>
      </styled.div>
    </Styled(Col)>
    <Styled(Col)
      lg={5}
      lgOffset={1}
      md={6}
    >
      <UpcomingTrainingSection
        curriculum={true}
        type="React bootcamp"
      />
    </Styled(Col)>
  </Row>
</Fragment>
`;
